Clarify product environment minimum for ECH

kunisen opened the pull request:
This is to address https://github.com/elastic/support-tech-lead/issues/1655 (internal ticket). 

## Preview - Before merge

- [deploy-manage/deploy/elastic-cloud/ec-customize-deployment-components.md](https://docs-v3-preview.elastic.dev/elastic/docs-content/pull/3307/deploy-manage/deploy/elastic-cloud/ec-customize-deployment-components)
- [deploy-manage/deploy/elastic-cloud/elastic-cloud-hosted-planning.md](https://docs-v3-preview.elastic.dev/elastic/docs-content/pull/3307/deploy-manage/deploy/elastic-cloud/elastic-cloud-hosted-planning)
- [deploy-manage/monitor/stack-monitoring/ece-ech-stack-monitoring.md](https://docs-v3-preview.elastic.dev/elastic/docs-content/pull/3307/deploy-manage/monitor/stack-monitoring/ece-ech-stack-monitoring)

## Preview - After merge

* https://www.elastic.co/docs/deploy-manage/deploy/elastic-cloud/elastic-cloud-hosted-planning
* https://www.elastic.co/docs/deploy-manage/deploy/elastic-cloud/ec-customize-deployment-components
* https://www.elastic.co/docs/deploy-manage/monitor/stack-monitoring/ece-ech-stack-monitoring

kunisen committed:
Clarify product environment minimum

https://github.com/elastic/support-tech-lead/issues/1655

kunisen committed:
Update elastic-cloud-hosted-planning.md

use sentence case

yetanothertw reviewed (approved):
Thanks @kunisen, these changes look good! I left a few minor suggestions for your consideration.

emrcbrn reviewed (approved):
Agreed with the minor changes suggest by @yetanothertw , LGTM otherwise. Thanks @kunisen

kunisen committed:
Update deploy-manage/deploy/elastic-cloud/ec-customize-deployment-components.md

Co-authored-by: Vlada Chirmicci <vlada.chirmicci@elastic.co>

kunisen committed:
Update deploy-manage/deploy/elastic-cloud/ec-customize-deployment-components.md

Co-authored-by: Vlada Chirmicci <vlada.chirmicci@elastic.co>

kunisen committed:
Update deploy-manage/monitor/stack-monitoring/ece-ech-stack-monitoring.md

Co-authored-by: Vlada Chirmicci <vlada.chirmicci@elastic.co>

kunisen committed:
revert the wording from "must" back to "should"

I accidentally committed a suggestion but as I explained in https://github.com/elastic/docs-content/pull/3307#discussion_r2404747219, we'd better use "should" as our verbal to avoid strong obligation.

kunisen commented:
@yetanothertw @emrcbrn thanks both! 

About whether we should use "must" or "should", please let's make it a little more clear - https://github.com/elastic/docs-content/pull/3307#discussion_r2404747219. Please let me know if you have different thoughts.
Thanks for your patience in advance. 

I will merge it once we reach to a common agreement with above.

yetanothertw commented:
Thank you for your patience with this, @kunisen and apologies for the confusion 💟 
`Should` works well as a recommendation, feel free to merge it 🚀

kunisen committed:
Update elastic-cloud-hosted-planning.md

small tweak about wording per https://github.com/elastic/docs-content/pull/3307#discussion_r2405415674

kunisen committed:
Merge branch 'main' into kunisen-docpr-stl-1655
